extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -576,14 +576,18 @@ export class AmpA4A extends AMP.BaseElement {
           // meeting the definition as opposed to waiting on the promise.
           if (this.delayAdRequestEnabled() &&
               !this.getResource().renderOutsideViewport()) {
-            return this.getResource().whenWithinRenderOutsideViewport();
+            return this.getResource().whenWithinRenderOutsideViewport().then(
+                () => this.tryExecuteRealTimeConfig_);
+          } else {
+            return this.tryExecuteRealTimeConfig_();
```

Comment:
Do we need to update the .then(() block here to include the correct return type?  E.g: /** @return {?./real-time-config-manager.rtcResponseDef} */

Reply:
done